Guide/DynamicProgramming/Abbreviation/README.md

```diff
@@ -0,0 +1,115 @@
+Category: Dynamic Programming
+Difficulty: Medium
+
+### Overview
+Read the link instructions,
+they are helpful when solving
+greater than eye deep
+
+Edge cases and complexities are covered below. Please read the problem first.
+
+## Algorithm
+### Overview
+This problem is complex because of the edge cases that exist. Some
+solutions don't need to worry about them, however when trying to maximize
+efficiency it becomes a problem. When I approached the problem I tried to
+use a double pointer method, iterating through both strings at the same time.
+
+The following example will show why this problem is tough to get right
+the first time, or second.
+
+It looks like this (credit to Orion for the edge case),
+- A = abcBBcA and B = ABBCA
+  - p1 (pointer one) = a | p2 (pointer two) = A
+    - a can be turned into A, so B is possible so far.
+  - p1 = b | p2 = B
+    - Another match
+  - p1 = c | p2 = B
+    - Move p1 forward (since c can be ignored and deleted).
+  - p1 = B | p2 = B
+    - Both are uppercase and they match, so move both pointers forward.
+  - p1 = B | p2 = C
+    - Oops! This situation causes a return of "NO", even though it should
+    print "YES". The problem is that the lowercase 'b' "uses" a slot that
+    should be used by the uppercase 'B'. Because lowercase characters can
+    be deleted but uppercase cannot, **every uppercase character must be
+    used by the algorithm.**
+
+So what's the solution? Well, either you can backtrack whenever you reach
+an uppercase character that could have been used up by a lowercase
+character previously. Or, you can create a kind of "memory" for the
+algorithm. A memory would need the following,
+- A unique ID for combinations of lowercase characters.
+- A way for uppercase characters to check if they could replace those
+lowercase characters that had been previously used.
+
+It should be possible to create a nearly unique ID for any combination of
+lowercase characters. By having a counter that increases as new
+characters are added to memory, and multiplying the numeric value
+of new characters by it as part of the hash, it will help guarantee a
+nearly unique ID for each sequence. Like the following example,
+- abc compared to bca where counter starts at one.
+  - a*counter + b*(counter+1) + c*(counter+2) = 1*1 + 2*2 + 3*3 = 14
+  - b*counter + c*(counter+1) + a*(counter+2) = 2*1 + 3*2 + 1*3 = 11
+
+This memory will be stored in an array, like a hash map. By adding
+in a modulus operator and by using a prime number for the
+modulus, we help guarantee that few IDs will conflict. The prime
+number will also match the size of the array used to store the
+memory.
+
+In using this memory we make it possible to get an O(M + N)
+time complexity for this problem, since there is no need
+to backtrack.
+
+Assumption: Although lowercase letters may be turned uppercase,
+every lowercase character remaining must be deleted. This is not
```

Comment:
It would probably help if I clicked the right line. I meant line 71; where you mention that it is _your_ assumption